### PR TITLE
[db] Fix indexExists

### PR DESCRIPTION
**Pull request by geropl**
## Description
Fixes `indexExists`.

This is still fallout from the version upgrade :slightly_smiling_face: 

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6978#issuecomment-982395188" title="Approved">geropl</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/6978#issuecomment-982395188" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/gitpod-db/src/typeorm/migration/helper/helper.ts
Comment (jankeromnes) [inline components/gitpod-db/src/typeorm/migration/helper/helper.ts:28]: Nit: Maybe it's safer to specify base 10?

```suggestion
    return Number.parseInt(countResult[0].cnt, 10) > 0; // for composite indexes this seems to return the number of columns involved
```

**Review by jankeromnes — APPROVED** [review]
Looks good to me, many thanks!

One minor suggestion, so holding, but feel free to ignore. 😊

/lgtm
/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 27be68bf7dbb2a037cf5dcbb38edf6fdb1e825b5</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6978#issuecomment-982395188" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6978#pullrequestreview-818697185" title="LGTM">jankeromnes</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/6978#issuecomment-982395188" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by geropl** [comment]
/unhold

I don't think that helps with the issue at hand. As general code-cleanup it's nice, but then we should be consequent and fix all usages, maybe in a separate PR.

**Comment by jankeromnes** [comment]
Fair enough. But thankfully, there are already a few code locations that are safe against surprise octals / hexadecimals:

https://github.com/gitpod-io/gitpod/blob/ae57b04985a120e9f52013f18da58d781b1d327f/components/server/src/github/github-context-parser.ts#L36

https://github.com/gitpod-io/gitpod/blob/ae57b04985a120e9f52013f18da58d781b1d327f/components/server/src/github/github-context-parser.ts#L50

https://github.com/gitpod-io/gitpod/blob/7c2e145b9d04df2618529b8e05aef2c55100ef69/components/dashboard/src/settings/Teams.tsx#L617

https://github.com/gitpod-io/gitpod/blob/7c2e145b9d04df2618529b8e05aef2c55100ef69/components/dashboard/src/settings/Teams.tsx#L682

Maybe this could be enforced by linting.

**Comment by jankeromnes** [comment]
(In this case, I case we're pretty safe until we reach composite indexes on 8+ rows 😅)

**Comment by geropl** [comment]
@jankeromnes You're absolutely right at the core; everyone should use the explicit base with `parseInt` every time. But as we're all humans, we all tend to forget about this from time to time. And what I really want to avoid is little discussions like this that consumes everyones time, again and again, just to fix a small part, that will get overwritten/broken again by another human being shortly after - experienced this too many times in the past to think that this is worth spending my lifetime on. :wink: 

I'm the first one to approve a PR with a linting rule and fix across project/the whole code base.

**Pull request merged**
